src/main/java/uk/gov/hmcts/dts/fact/services/admin/AdminCourtAddressService.java

```diff
@@ -1,33 +1,43 @@
 package uk.gov.hmcts.dts.fact.services.admin;
 
+import org.apache.commons.lang.StringUtils;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.CollectionUtils;
 import uk.gov.hmcts.dts.fact.entity.Court;
 import uk.gov.hmcts.dts.fact.exception.NotFoundException;
+import uk.gov.hmcts.dts.fact.mapit.MapitData;
 import uk.gov.hmcts.dts.fact.model.admin.CourtAddress;
 import uk.gov.hmcts.dts.fact.repositories.CourtAddressRepository;
 import uk.gov.hmcts.dts.fact.repositories.CourtRepository;
+import uk.gov.hmcts.dts.fact.services.MapitService;
 import uk.gov.hmcts.dts.fact.services.admin.list.AdminAddressTypeService;
+import uk.gov.hmcts.dts.fact.services.validation.ValidationService;
 import uk.gov.hmcts.dts.fact.util.AddressType;
 
-import java.util.Comparator;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
 
 import static java.util.stream.Collectors.toList;
 
 @Service
+@SuppressWarnings("PMD.DataflowAnomalyAnalysis")
 public class AdminCourtAddressService {
     private final CourtRepository courtRepository;
     private final CourtAddressRepository courtAddressRepository;
     private final AdminAddressTypeService addressTypeService;
+    private final AdminService adminService;
+    private final MapitService mapitService;
+    private final ValidationService validationService;
 
     @Autowired
-    public AdminCourtAddressService(final CourtRepository courtRepository, final CourtAddressRepository courtAddressRepository, final AdminAddressTypeService addressTypeService) {
+    public AdminCourtAddressService(final CourtRepository courtRepository, final CourtAddressRepository courtAddressRepository, final AdminAddressTypeService addressTypeService, AdminService adminService, MapitService mapitService, ValidationService validationService) {
```

Comment:
Did you want to add final's onto the other three parameters here too?

AdminService adminService, MapitService mapitService, ValidationService validationService

Reply:
Ohh yes I should do really. The new ones are auto added by Intellij. I will add 'final' there